Sources/NIOCore/ChannelHandler.swift

```diff
@@ -343,3 +343,19 @@ extension RemovableChannelHandler {
         context.leavePipeline(removalToken: removalToken)
     }
 }
+
+/// A `OutboundBufferedBytesAuditableChannelHandler` is a `ChannelHandler` that 
+/// audits and reports the number of bytes buffered for outbound direction.
+public protocol OutboundBufferedBytesAuditableChannelHandler {
+    /// Returns the number of bytes buffered in the channel handler, which are queued to be sent to
+    /// the next outbound channel handler.
+    func auditOutboundBufferedBytes() -> Int
```

Comment:
Minor nit: these should probably be computed properties, not functions.

Reply:
Only if we want to add an `O(1)` requirement. I don't we think we want though because some ChannelHandlers have `n` child ChannelHandlers which means they wouldn't be able to conform, strictly speaking. 

It would also be unnecessarily different from the other overloads that take an argument.

Reply:
Why is this now a computed property? Do we really want it to be O(1)? We should at least document it to be potentially O(n) where n is the number of child channels to be in line with the API Design Guidelines:
> Document the complexity of any computed property that is not O(1). People often assume that property access involves no significant computation, because they have stored properties as a mental model. Be sure to alert them when that assumption may be violated.
> https://www.swift.org/documentation/api-design-guidelines/#conventions

cc: @Lukasa